src/Analysis.cpp

```diff
@@ -290,6 +303,33 @@ void doAliasReasoning(Instruction *instruction,
         }
       }
     }
+  } else if (AllocaInst *allocate = dyn_cast<AllocaInst>(instruction)) {
+    if (llvm::StructType *structType =
+            llvm::dyn_cast<llvm::StructType>(allocate->getAllocatedType())) {
+      /*
+      I cannot find documentation that states how struct names appear in the IR,
+      but it seems that they typically look like "struct.[C struct name]"
+      TODO: prove/disprove that struct names are formatted as "struct.[C struct
+      name]"
+      */
+      std::string structName = structType->getName();
+      structName = sliceString(structName, structName.find_last_of('.') + 1,
+                               structName.size() - 1);
+      int numFields = structType->getNumElements();
+      for (int i = 0; i < numFields; i++) {
+
+        Annotation *anno = annotationHandler.getStructAnnotation(
+            structName, std::to_string(i));
```

Comment:
I'm worried that this might not work properly if there are two files being analyzed that each contain a struct that happens to have the same name. That case will come up in real code, so this needs to be keyed not just on the name of the struct, but also which file it is declared in (or something else that uniquely identifies a particular struct, which the name does not do).